variants/frontend-base/template.rb

```diff
@@ -57,7 +57,57 @@
 EO_IMG_EXAMPLE
 gsub_file "app/views/layouts/application.html.erb", "<body>", body_open_tag_with_img_example, force: true
 
+# Setup Sentry
+# ############
+
+run "yarn add @sentry/browser"
+run "yarn add dotenv-webpack -D"
+
+gsub_file "config/webpack/environment.js",
+  "module.exports = environment",
+  <<~'EO_JS'
+    const Dotenv = require('dotenv-webpack');
+
+    environment.plugins.prepend('Dotenv', new Dotenv());
+
+    module.exports = environment;
+  EO_JS
+
+append_to_file "app/frontend/packs/application.js" do
+  <<~'EO_JS'
+
+    // Initialize Sentry Error Reporting
+    //
+    // Import all your application's JS after this section because we need Sentry
+    // to be initialized **before** we import any of our actual JS so that Sentry
+    // can report errors from it.
+    //
+    import * as Sentry from '@sentry/browser';
+    Sentry.init({
+      dsn: process.env.SENTRY_DSN,
+      environment: process.env.SENTRY_ENV || process.env.RAILS_ENV
```

Comment:
FYI @G-Rath @joshmcarthur now using RAILS_ENV as fallback to mimic the Ruby behaviour

Reply:
fwiw `NODE_ENV` is the equivalent to `RAILS_ENV`, but I don't think it matters too much.

Reply:
Hmmm good point. I think there is a risk that `NODE_ENV` get not set properly when rails runs webpacker to compile the assets - `RAILS_ENV` seems safer (in the context of a Rails app)

Reply:
[`NODE_ENV` is set by webpack based on the `mode`](https://webpack.js.org/guides/production/#specify-the-mode), which also controls some of the output of webpack itself, so if it's not being set to `production` at the right places (by webpacker or otherwise) we've got bigger problems already.